### PR TITLE
store: remove stores that have no region before balance

### PR DESCRIPTION
**Pull request by xzhangxian1008**
### What problem does this PR solve?

Issue Number: ref #52313

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: store/copr/batch_coprocessor.go [hunk 1]
```diff
@@ -329,8 +342,9 @@ func balanceBatchCopTask(ctx context.Context, kvStore *kvStore, originalTasks []
 			storeTaskMap[taskStoreID] = batchTask
 		}
 	} else {
-		stores := cache.RegionCache.GetTiFlashStores(tikv.LabelFilterNoTiFlashWriteNode)
-		aliveStores := filterAliveStores(ctx, stores, ttl, kvStore)
+		usedStores := getUsedStores(cache, usedTiFlashStoresMap)
+		logutil.BgLogger().Info("detecting available mpp stores")
```
Comment (windtalker) [inline store/copr/batch_coprocessor.go:346]: Looks like this log is duplicated since `filterAliveStores` will also record this log

**Review by zanmato1984 — APPROVED** [review]

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**